adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -720,8 +827,31 @@ public void acquireTokenByRefreshToken(String refreshToken, String clientId, Str
      */
     public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
         if (requestCode == AuthenticationConstants.UIRequest.BROWSER_FLOW) {
-            final AcquireTokenRequest acquireTokenRequest = new AcquireTokenRequest(mContext, this);
-            acquireTokenRequest.onActivityResult(requestCode, resultCode, data);
+
+            if (data == null) {
+                // If data is null, RequestId is unknown. It could not find
+                // callback to respond to this request.
+                Logger.e(TAG, "onActivityResult BROWSER_FLOW data is null.", "",
+                        ADALError.ON_ACTIVITY_RESULT_INTENT_NULL);
```

Comment:
line 829 is also duplication, how about i file an issue to clean up on Activity result?

---
In reply to: [76351142](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/703#discussion_r76351142) [](ancestors = 76351142)